toolkit/tools/imagegen/configuration/parse_partition.go

```diff
@@ -87,7 +87,28 @@ func processDisk(inputDiskValue string) (err error) {
 			disks = append(disks, Disk{})
 		}
 
-		disks[latestDiskIndex].PartitionTableType = PartitionTableTypeGpt
+		// In kickstart installation scenario, the partition table type is set to
+		// MBR by default. The Anaconda installer has this config "--gpt" that indicates
+		// whether the users prefer creation of GPT disk label or not. The value of "--gpt"
+		// is a bool where "True" indicates using GPT and "False" if not, which means using MBR.
+		// This config is set as a boot option within /proc/cmdline, which will be parsed by anaconda
+		// during installation process. Thus, Mariner will also pick the same design to reach compatibility
+		// with kickstart scenario
+
+		// Please note that this code is only executed during kickstart installation, when "IsKickStartBoot" is set to true.
+		// Mariner installer currently does not allow direct specification of disk and partition layout within
+		// the image config file for kickstart installation. So any disk/partition setting you make in the image config file
+		// will be overwritten if you enable kickstart installation mode.
+		isGPTPartitionTable, err := GetKernelCmdLineValue("--gpt")
+		if err != nil {
+			return err
+		}
+
+		if strings.TrimSpace(isGPTPartitionTable) == "True" {
+			disks[latestDiskIndex].PartitionTableType = PartitionTableTypeGpt
+		} else {
+			disks[latestDiskIndex].PartitionTableType = PartitionTableTypeMbr
+		}
```

Comment:
Considering the size of this comment and how this logic is nicely limited, I'd suggest extracting this to a separate function like `setPartitionType` passing the disk by reference.

Reply:
Nice point! Fixed.

Reply:
Holly crap, I had no idea that this whole file is using global variables. A bit scary but not related to this PR.;)